lib/portlayer/event/collector/vsphere/collector.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2016 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vsphere
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/vmware/vic/lib/portlayer/event/events"
+	"github.com/vmware/vic/pkg/vsphere/session"
+
+	vmwEvents "github.com/vmware/govmomi/event"
+	"github.com/vmware/govmomi/vim25/types"
+
+	log "github.com/Sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+const (
+	name = "vSphere Event Collector"
+)
+
+type EventCollector struct {
+	vmwManager *vmwEvents.Manager
+	mos        monitoredCache
+	callback   func(events.Event)
+}
+
+type monitoredCache struct {
+	mu sync.RWMutex
+
+	mos map[string]types.ManagedObjectReference
+}
+
+func NewCollector(session *session.Session, objects ...string) *EventCollector {
```

Comment:
Can we use Client directly here instead of session? This appears to be the only dependency on session in the collector.